libfwupdplugin/fu-smbios.c

```diff
@@ -89,6 +93,80 @@ fu_smbios_convert_dt_string (FuSmbios *self, guint8 type, guint8 offset,
 	fu_smbios_convert_dt_value (self, type, offset, item->strings->len);
 }
 
+#if defined (__FreeBSD__) || defined (__DragonFly__)
+static void
+fu_smbios_kenv_sysctl_string (FuSmbios *self, guint8 type, guint8 offset,
+				const gchar *buf, gsize bufsz)
+{
+	FuSmbiosItem *item = g_ptr_array_index (self->items, type);
+
+	/* add to strtab */
+	g_ptr_array_add (item->strings, g_strndup (buf, bufsz));
+	fu_smbios_convert_dt_value (self, type, offset, item->strings->len);
+}
+
+static gboolean
+fu_smbios_kenv_lookup (const char* sminfo, gchar *buf, gsize bufsz)
+{
+	g_debug ("DMI request for %s", sminfo);
+	if (kenv (KENV_GET, sminfo, buf, bufsz - 1) == -1) {
+		g_debug ("Cannot get %s", sminfo);
+		return FALSE;
+	} else {
```

Comment:
you don't need this if you're returning on the other clause.

Reply:
Deleted.